pwndbg/commands/context.py

```diff
@@ -580,7 +580,7 @@ def context_backtrace(with_banner=True, target=sys.stdout, width=None):
     for i in range(backtrace_lines-1):
         try:
             candidate = oldest_frame.older()
-        except gdb.MemoryError:
```

Comment:
Out of curiosity, what's the gdb.error that can happen here? Any chance we can commit this separately to persist the reason of this or at least add a comment why this or the other error?

Reply:
I test it on `Ubuntu 22.04`.
```sh
apt install qemu-user
wget "https://github.com/M0ngi/CTF-Writeups/raw/main/2022/Hacker's%20Playground%202022/sources/riscy/deploy/target"
chmod +x target
python3 -c 'import sys; sys.stdout.buffer.write(b"AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAV\x22\x04\x00\x00\x00\x00\x00AAAAAAAAAAAAAAAAAAAAAAAA\xcc\xdd\x01\x00\x00\x00\x00\x00AAAAAAAA\x82\x17\x04\x00\x00\x00\x00\x00aaaabaaa\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x08\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00iaaajaaakaaalaaamaaanaaaoaaapaaa?\x00\x00\x00\x00\x00\x00\x004\x04\x01\x00\x00\x00\x00\x00")' | qemu-riscv64 -g 9000 ./target
gdb-multiarch -ex 'set exception-verbose on' -ex 'target remote localhost:9000' -ex 'b *0x42256' -ex 'c' ./target
```
I got the error:
```
Traceback (most recent call last):
  File "/home/loop/pwndbg/pwndbg/commands/__init__.py", line 129, in __call__
    return self.function(*args, **kwargs)
  File "/home/loop/pwndbg/pwndbg/commands/__init__.py", line 223, in _OnlyWhenRunning
    return function(*a, **kw)
  File "/home/loop/pwndbg/pwndbg/commands/context.py", line 272, in context
    result[target].extend(func(target=out,
  File "/home/loop/pwndbg/pwndbg/commands/context.py", line 582, in context_backtrace
    candidate = oldest_frame.older()
gdb.error: PC not saved
```

Reply:
@disconnect3d What should we do with it?

Reply:
Hm I'm pretty sure I was responding here already but I guess my comment was not send.

We should just add a comment about it like this:
```
# We catch gdb.error in case of a `gdb.error: PC not saved` case
except (gdb.MemoryError, gdb.error):
```

Reply:
@disconnect3d Thanks for pointing it out directly.